tensorflow/compiler/xla/service/gpu/amdgpu_compiler.h

```diff
@@ -79,6 +79,8 @@ class AMDGPUCompiler : public LLVMCompiler {
   static const char* kDataLayout;
 
  private:
+  se::Platform::Id platform_id_;
```

Comment:
Does it make sense if the id is a `static const` member? Just to be safer, I feel like The `AMDGPUCompiler` should only have hte id as `kROCmPlatformId` under any circumstances.

Reply:
The eventual goal is to merge two subclasses so I feel the field should be PlatformId

Reply:
And we populate the field thru ctor

Reply:
Right... I was just looking at the code as is. When they are merged, there need a way to populate both compilers.